enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,435 @@
+
+In order to address the existing plain provisioner limitation, this proposal also recommends expanding the plain+v0 bundle format, and enabling users to define health checks through a dedicated metadata file. A valid metadata file may be named "bundle.yaml" so clients can know where to find this file.
+
+For phase 0, the registry+v1 provisioner would be responsible for automatically generating this metadata file for PO packages. Future phases may introduce support for defining PO packages using OLM's plain bundle format, which will require operator authors to explicitly define this metadata file themselves.
+
+These metadata-level health checks can be expressed using JSONPath templates to declaratively define custom probe logic. When the plain provisioner is processing unpacked Bundle contents, and a manifests file has been defined, it’s responsible for serializing the values defined in the metadata file, and evaluating any defined custom probes. When an individual probe evaluation fails, that state will be reflected in the BundleDeployment’s status.
+
+In the case that a probe was misconfigured, an admin may elect to toggle this behavior off entirely. The ProvisionerClass API, which doesn’t exist today, would be a candidate resource for providing this escape hatch behavior. Without that API in place today, the `plain.core.rukpak.io/enable-probes` BundleDeployment annotation key may serve as a short term solution. The plain provisioner implementation would be responsible for respecting that annotation key.
+
+Injecting this annotation may require changes to the top-level PlatformOperators resource to introduce a toggle field to control this behavior. When the PO manager sees this toggle has been disabled, it's responsible for propagating this configuration as an annotation to the set of BundleDeployment resources it manages.
+
+When reconciling a PO custom resource, the PO manager may need to propagate the underlying BundleDeployment's status back to the PO custom resource that's currently being processed. A separate controller may be responsible for watching, and listing the PO custom resource, and relaying that individual PO status information back to the aggregate ClusterOperator that controller is managing.
+
+### Workflow Description
+
+```mermaid
+sequenceDiagram
+actor Admin
+participant Installer
+participant Bootstrap
+participant CVO
+participant ClusterOperator
+participant PO manager
+Installer->>Admin: Creates the Kubernetes manifests
+Installer->>PO custom resource: Generates
+Admin-->>Installer: Edits Kubernetes manifests
+Installer->>Bootstrap: Creates
+Bootstrap->>CVO: Deploys
+CVO->>PO manager: Deploys
```

Comment:
i'm curious if this will be happening in the bootstrap or once the control plane has been created or perhaps some time later?

Reply:
Assumption thus far is that the POM would be deployed like any other SLO.  It would definitely need to be after we have a functioning control plane.

Reply:
cool, i'm very curious if we would be able to utilize the PO to install a MAPI operator which could then roll out the compute plane. like, once the control plane has been created, install the mapi operators, then grow the cluster through that mechanism. i know this is a little different than an agent based workflow, but i'm just thinking about options. thanks!

Reply:
I had imagined we'd continue to deploy this POM at the same run-level (if that's still a thing in the CVO?) as the rest of the OLM stack, which should be sufficient to meet phase 0 requirements. It's unclear whether continuing that approach has any significant implications for future phase designs when attempting to onboard more involved operators to OLM. Maybe we can track this MAPI use case $somewhere while we continue to iterate on phase designs?

Reply:
Same comment as above: https://github.com/openshift/enhancements/pull/1170/files#r932557842. Any preferences for how we can track the potential MAPI/CCM use cases here? I think it could either live as an open question, or an explicit non-goal, or something we track downstream in a jira epic.